plugins/modules/cloud/misc/proxmox_nic.py

```diff
@@ -0,0 +1,344 @@
+      - When not specified this module will keep the MAC address the same when changing an existing interface.
+    type: str
+  model:
+    description:
+      - The NIC emulator model.
+    type: str
+    choices: ['e1000', 'e1000-82540em', 'e1000-82544gc', 'e1000-82545em', 'i82551', 'i82557b', 'i82559er', 'ne2k_isa', 'ne2k_pci', 'pcnet',
+              'rtl8139', 'virtio', 'vmxnet3']
+    default: virtio
+  mtu:
+    description:
+      - Force MTU, for C(virtio) model only. Set to C(1) to use the bridge MTU.
+      - Value should be C(1 ≤ n ≤ 65520).
+    type: int
+  name:
+    description:
+      - Specifies the VM name. Only used on the configuration web interface.
+      - Required only for I(state=present).
+    type: str
+  queues:
+    description:
+      - Number of packet queues to be used on the device.
+      - Value should be C(0 ≤ n ≤ 16).
+    type: int
+  rate:
+    description:
+      - Rate limit in MBps (MegaBytes per second) as floating point number.
+    type: float
+  state:
+    description:
+      - Indicates desired state of the NIC.
+    type: str
+    choices: ['present', 'absent']
+    default: present
+  tag:
+    description:
+      - VLAN tag to apply to packets on this interface.
+      - Value should be C(1 ≤ n ≤ 4094).
+    type: int
+  trunks:
+    description:
+      - List of VLAN trunks to pass through this interface.
+    type: list
+    elements: int
+  vmid:
+    description:
+      - Specifies the instance ID.
+    type: int
+extends_documentation_fragment:
+  - community.general.proxmox.documentation
+'''
+
+EXAMPLES = '''
+- name: Create NIC net0 targeting the vm by name
+  community.general.proxmox_nic:
+    api_user: root@pam
+    api_password: secret
+    api_host: proxmoxhost
+    name: my_vm
+    interface: net0
+    bridge: vmbr0
+    tag: 3
+
+- name: Create NIC net0 targeting the vm by id
+  community.general.proxmox_nic:
+    api_user: root@pam
+    api_password: secret
+    api_host: proxmoxhost
+    vmid: 103
+    interface: net0
+    bridge: vmbr0
+    mac: "12:34:56:C0:FF:EE"
+    firewall: true
+
+- name: Delete NIC net0 targeting the vm by name
+  community.general.proxmox_nic:
+    api_user: root@pam
+    api_password: secret
+    api_host: proxmoxhost
+    name: my_vm
+    interface: net0
+    state: absent
+'''
+
+RETURN = '''
+vmid:
+  description: The VM vmid.
+  returned: success
+  type: int
+  sample: 115
+msg:
+  description: A short message
+  returned: always
+  type: str
+  sample: "Nic net0 unchanged on VM with vmid 103"
+'''
+
+try:
+    from proxmoxer import ProxmoxAPI
+    HAS_PROXMOXER = True
+except ImportError:
+    HAS_PROXMOXER = False
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils._text import to_native
+
+
+def get_vmid(proxmox, name):
+    return [vm['vmid'] for vm in proxmox.cluster.resources.get(type='vm') if vm.get('name') == name]
```

Comment:
This comment is not unique to this module, but for idempotency purposes I think it would be better to fail when multiple guests are returned given that PVE does not guarantee uniqueness of names across clusters or individual nodes. However this is consistent with the existing modules.